packages/docusaurus-theme-redoc/src/theme/Redoc/ServerStyles.tsx

```diff
@@ -39,8 +39,9 @@ export function ServerStyles({
   specProps: SpecProps;
   lightThemeOptions: RedocRawOptions;
   darkThemeOptions: RedocRawOptions;
+  absolute: boolean
 }) {
-  const fullUrl = useBaseUrl(specProps.url, { absolute: true });
+  const fullUrl = specProps.normalizeUrl ? useBaseUrl(specProps.url, { absolute: true }) : specProps.url;
```

Comment:
No conditional hooks to follow rules of react
```suggestion
  const absoluteUrl = useBaseUrl(specProps.url, { absolute: true });
  const fullUrl = specProps.normalizeUrl ? absoluteUrl : specProps.url;
```